src/gangliad/statsd.cpp

```diff
@@ -598,6 +691,29 @@ StatsD::initAndReconfig(char const *service_name)
 	formatstr(param_name,"%s_REQUIREMENTS",service_name);
 	param(m_requirements,param_name.c_str());
 
+	m_reset_metrics_filename.clear();
+	formatstr(param_name,"%s_WANT_RESET_METRICS",service_name);
+	if (param_boolean(param_name.c_str(),false)) {
+		formatstr(param_name,"%s_RESET_METRICS_FILE",service_name);
+		param(m_reset_metrics_filename,param_name.c_str());
+		
+		if (!m_reset_metrics_filename.empty()) {
+			// If filename from the user is a relative path, stick it in SPOOL dir
+			if ( !IS_ANY_DIR_DELIM_CHAR(m_reset_metrics_filename[0]) ) {
+				std::string fname = m_reset_metrics_filename;
+				std::string dirname;
+				param(dirname,"SPOOL");
+				dircat(dirname.c_str(),fname.c_str(),m_reset_metrics_filename);
```

Comment:
I think if `m_reset_metrics_filename` is `../../foo` the resulting file path is `$(SPOOL)/../../foo`. Is that the intended behavior?

Reply:
Good thinking. The idea that if the config knob does not specify a fully-qualified path, then the specified path is relative to the $(SPOOL) directory.  I was only thinking about the case where the admin specified a file name instead of a relative path, but I don't see a problem in allowing it to be either file or relative path.

Reply:
I think that is reasonable.